### ISSUE TITLE
Unable to order a dedicatedhost from SLCLI for a GPU Flavor

### ISSUE THREAD

**Issue by FernandoOjeda** [softlayer/softlayer-python#1022]
Unable to order a dedicatedhost from slcli for a GPU flavor.

slcli dedicatedhost create -H E2ETestHost1 -D e2eperftest.com -f 56_CORES_X_484_RAM_X_1_5_TB_X_2_GPU_P100 -d dal13

Receive the below error when I submit an order request.

SoftLayerAPIError(SoftLayer_Exception_Public): Unable to find a price for the dedicated host configuration provided.

The slcli to get the routers available for the dedicatedHost item use the following structure:

SoftLayer_Virtual_DedicatedHost::getAvailableRouters

```
{
      "cpuCount": 56,
      "diskCapacity": 1200,
      "memoryCapacity": 360,
      "datacenter": {
      	"id": 1854895
      }

```
But for dedicatedHost GPU flavor has to add the following structure:

```
{
      "cpuCount": 56,
      "diskCapacity": 1200,
      "memoryCapacity": 360,
      "datacenter": {
      	"id": 1854895
      },
      "pciDevices": [ 
      	{ 
      		"hardwareComponentModel": 
      		   { 
      		   	   "hardwareGenericComponentModel":
			               {"hardwareComponentType":
					{"keyName":"GPU"
					}
				}
      		   	
      		   } 
      	}, 
      	{ 
      		"hardwareComponentModel": 
      		   { 
      		   	   "hardwareGenericComponentModel":
			               {"hardwareComponentType":
					{"keyName":"GPU"
					}
				}
        	 } 
       	} 
      ] 
    }

```
Environment Information
Operating System: 
softlayer-python version (`slcli --version`): version 5.5.0

**Comment by kwienken** [comment]
@FernandoOjeda I don't think using the hardwareComponentType alone is enough since not all dedicated hosts use the same GPU model. You will need to get the model from the bundled GPU product. The bundled product also specifies the number of GPUs.

**Comment by allmightyspiff** [comment]
@kwienken given just the flavor 56_CORES_X_484_RAM_X_1_5_TB_X_2_GPU_P100 how would we get the needed structure to call SoftLayer_Virtual_DedicatedHost::getAvailableRouters?

**Comment by kwienken** [comment]
You'll need to get the product by keyName and use the mask ```
mask[
    capacity,
    units,
    hardwareGenericComponentId,
    itemCategory[categoryCode],
    bundleItems[
        capacity,
        units,
        hardwareGenericComponentId,
        itemCategory[categoryCode],
    ]
]```

In the dedicated host template you'll set:
* cpuCount equal the capacity value of the top-level product
* diskCapacity equal to the capacity value of the dedicated_host_disk bundled product
* memoryCapacity equal to the capacity value of the dedicated_host_ram bundled product
* pciDevices empty array if no dedicated_host_gpu bundled product(s) found or an array of hardwareComponentModel objects equaling the capacity of the dedicated_host_gpu bundled product with each object having hardwareGenericComponentModelId equal to the hardwareGenericComponentId on the product

**Comment by FernandoOjeda** [comment]
@kwienken The structure that I am using is the same structure that the portal control uses, it does not use any hardwareGenericComponentId but I will try adding the respective identifier.

Thank you

**Comment by allmightyspiff** [comment]
This fix is in 5.5.2
